src/app/services/generic-node-api/generic-node-api.ts

```diff
@@ -1,30 +1,34 @@
 import { Injectable } from '@angular/core';
 import { Http, Response } from '@angular/http';
 
-import { Observable } from 'rxjs/Observable';
+import { Observable } from 'rxjs/Rx';
 import 'rxjs/add/operator/catch';
 import 'rxjs/add/operator/map';
 
-
 @Injectable()
-export class CurrenciesService {
+export class GenericNodeApiService {
 
-  private currenciesApiUrl = 'http://localhost:4200/assets/coins.json';
+  baseApiUrl: string = 'http://localhost:3000/api/'
 
   constructor(private http: Http) { }
 
-  getCurrencies(): Observable<Object[]> {
-    return this.http.get(this.currenciesApiUrl)
+  getRequest(apiRouteUrl): Observable<Object[]> {
+    return this.http.get(this.baseApiUrl + apiRouteUrl)
                     .map(this.extractData)
-                    .catch(this.handleError);
+                    .catch(this.handleError)
   }
 
   private extractData(res: Response) {
-    let body = res.json();
-    return body.data || { };
+    try{
+      let body = res.json();
+      return body.result || { };
+    } catch (e) {
+      console.error('Error extracting api data: ' + e )
```

Comment:
do we need to return something here if there is an error as well as logging it to the console?